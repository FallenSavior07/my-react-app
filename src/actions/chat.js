export const SET_CURRENT_CHAT = 'CHATS::CHANGE_CHAT';

export const setCurrentChat = (chat) => ({
	type: SET_CURRENT_CHAT,
	payload: {
		id: chat.id,
	},
})
