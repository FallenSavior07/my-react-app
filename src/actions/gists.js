import { API_URL_PUBLIC } from "../components/App/constants";

export const GET_GISTS = "GISTS::GET_GISTS";
export const SET_IDLE_STATUS = "GISTS::SET_IDLE_STATUS";
export const SET_LOADING_STATUS = "GISTS::SET_LOADING_STATUS";
export const SET_ERROR_STATUS = "GISTS::SET_ERROR_STATUS";

export const getGists = (gists) => ({
    type: GET_GISTS,
    payload: {
        gists
    }
});

export const setIdleStatus = () => ({
    type: SET_IDLE_STATUS,
})

export const setLoadingStatus = () => ({
    type: SET_LOADING_STATUS,
})

export const setErrorStatus = () => ({
    type: SET_ERROR_STATUS,
})

export const fetchGists = () => async (dispatch) => {
        dispatch(setLoadingStatus());
        try {
            const response = await fetch(API_URL_PUBLIC);
            if (!response.ok) {
                throw new Error(`Запрос не выполнен со статусом: ${response.status}`);
            }
            const result = await response.json();
            dispatch(getGists(result));
            dispatch(setIdleStatus());
        }
        catch (err) {
            console.error(err);
            dispatch(setErrorStatus());
        }
    }