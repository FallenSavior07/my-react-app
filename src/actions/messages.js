import { currentDate } from '../shared/currentDate';
import firebase from 'firebase';

export const ADD_MESSAGE = 'MESSAGES::ADD_MESSAGE';
export const CLEAR_MESSAGES_BY_CHATID = 'MESSAGES::CLEAR_MESSAGES_BY_CHATID';

export const addMessage = (chatId, message) => ({
	type: ADD_MESSAGE,
	payload: {
		chatId,
		message
	}
});

export const clearMessagesByChatId = (chatId) => ({
	type: CLEAR_MESSAGES_BY_CHATID,
	payload: {
		chatId
	}
});

export const addMessageWithThunk = (chatId, message) => {
	return () => {
		firebase.database()
			.ref('messages')
			.child(chatId)
			.push(message)

		let timer = setTimeout(() => {
			firebase.database()
				.ref('messages')
				.child(chatId)
				.push({
					id: `message${Date.now()}`,
					author: 'Робот Василий',
					text: `Привет, ${message.author}! Как дела?`,
					date: currentDate(),
				})

			clearTimeout(timer);
		}, 1500)
	}
}

export const subscribeOnMessagesChangings = (chatId) => {
	return (dispatch, getState) => {
		firebase.database()
			.ref('messages')
			.child(chatId)
			.on('child_added', (snapshot) => {
				dispatch(addMessage(chatId, snapshot.val()))
			});

		firebase.database()
			.ref('messages')
			.child(chatId)
			.on('child_changed', (snapshot) => {
				dispatch(addMessage(chatId, snapshot.val()))
			})
	}
}