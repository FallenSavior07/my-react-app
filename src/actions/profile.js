export const CHANGE_NAME = 'PROFILE::CHANGE_NAME';
export const CHANGE_IS_ONLINE = 'PROFILE::CHANGE_IS_ONLINE';
export const CHANGE_IS_AUTHED = 'PROFILE::CHANGE_IS_AUTHED';

export const changeName = (name) => ({
	type: CHANGE_NAME,
	payload: {
		name
	},
});

export const changeIsOnline = (isOnline) => ({
	type: CHANGE_IS_ONLINE,
	payload: {
		isOnline
	}
});

export const changeIsAuthed = (isAuthed) => ({
	type: CHANGE_IS_AUTHED,
	payload: {
		isAuthed,
	},
})
