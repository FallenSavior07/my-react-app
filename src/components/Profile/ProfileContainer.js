import '../../css/style.css';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Profile from './Profile';
import { changeIsOnline, changeName } from '../../actions/profile';
import { profileSelector } from '../../selectors/profile';
import firebase from 'firebase';

export default function ProfileContainer() {
	const { age, name, isOnline } = useSelector(profileSelector);

	const dispatch = useDispatch();

	const handleChangeIsOnline = (event) => {
		dispatch(changeIsOnline(event.target.checked))
	}

	const handleChangeName = (newUserName) => {
		dispatch(changeName(newUserName));
	}

	const handleSignOut = (event) => {
		event.preventDefault()
		firebase.auth().signOut()
	}

	return (
		<Profile
			age={age}
			name={name}
			isOnline={isOnline}
			handleChangeIsOnline={handleChangeIsOnline}
			handleChangeName={handleChangeName}
			handleSignOut={handleSignOut}
		/>
	)
}