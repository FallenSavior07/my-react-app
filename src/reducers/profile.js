import { CHANGE_NAME, CHANGE_IS_ONLINE } from '../actions/profile';

const initialState = {
	name: 'Вадим',
	age: 29,
	isOnline: true
}

export default function profileReducer(state = initialState, action) {
	switch (action.type) {
		case CHANGE_NAME: {
			return {
				...state,
				name: action.payload.name,
			}
		}
		case CHANGE_IS_ONLINE: {
			return {
				...state,
				isOnline: action.payload.isOnline
			}
		}
		default:
			return state
	}
}
