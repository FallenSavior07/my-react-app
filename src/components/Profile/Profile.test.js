import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import { changeIsOnline, changeName } from '../../actions/profile';
import { render } from "@testing-library/react";
import ProfileContainer from './ProfileContainer';
import { Provider } from 'react-redux';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);


describe('Profile test', () => {
	it('matches snapshot online', () => {
		const store = mockStore(() => ({ profile: {} }));

		const component = render(
			<Provider store={store}>
				<ProfileContainer />
			</Provider>
		);

		expect(component).toMatchSnapshot();
	});


	it('dispatches changeName', () => {
		let state = {};
		const store = mockStore(() => state);
		store.dispatch(changeName('John'));

		const actions = store.getActions().map(({ type, payload }) => ({ type, payload }));
		const expectedActions = [{ type: 'PROFILE::CHANGE_NAME', payload: { name: 'John' } }];
		expect(actions).toEqual(expectedActions);
	});

	it('dispatches changeIsOnline', () => {
		let state = {};
		const store = mockStore(() => state);
		store.dispatch(changeIsOnline('false'));

		const actions = store.getActions().map(({ type, payload }) => ({ type, payload }));
		const expectedActions = [{ type: 'PROFILE::CHANGE_IS_ONLINE', payload: { isOnline: 'false' } }];
		expect(actions).toEqual(expectedActions);
	});
})