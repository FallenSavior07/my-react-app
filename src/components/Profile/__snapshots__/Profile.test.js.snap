// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile test matches snapshot online 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="app__main main"
      >
        <section
          class="profile container"
        >
          <h2
            class="profile__title visually-hidden"
          >
            Profile page
          </h2>
          <div
            class="profile__inner"
          >
            <div
              class="profile__info info"
            >
              <h4
                class="info__title"
              >
                Информация о пользователе
              </h4>
              <img
                alt="avatar"
                class="info__image"
                height="120"
                src="avatar.png"
                width="120"
              />
              <p
                class="info__name"
              >
                Имя: 
              </p>
              <p
                class="info__age"
              >
                Возраст: 
              </p>
            </div>
            <hr
              class="profile__line"
            />
            <div
              class="profile__settings settings"
            >
              <h4
                class="settings__title"
              >
                Настройки
              </h4>
              <label
                class="MuiFormControlLabel-root settings__online-status"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiCheckbox-root Component-root-1"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <input
                      class="PrivateSwitchBase-input-6"
                      data-indeterminate="false"
                      name="checkedG"
                      type="checkbox"
                      value=""
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                >
                  Показывать статус "в сети"?
                </span>
              </label>
              <div
                class="settings__name change-name-form"
              >
                <p
                  class="change-name-form__label"
                >
                  Изменить имя
                </p>
                <form
                  action="#"
                  class="change-name-form__input message-form__button"
                >
                  <div
                    class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-8 message-form__input"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused Mui-required Mui-required"
                      data-shrink="true"
                      for="outlined-basic"
                      id="outlined-basic-label"
                    >
                      Введите новое имя...
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                      >
                         
                        *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root Mui-focused Mui-focused MuiInputBase-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input"
                        id="outlined-basic"
                        required=""
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="PrivateNotchedOutline-legendLabelled-11 PrivateNotchedOutline-legendNotched-12"
                        >
                          <span>
                            Введите новое имя...
                             *
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-7 message-form__button MuiButton-containedPrimary"
                    tabindex="0"
                    type="submit"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Отправить
                      <span
                        class="MuiButton-endIcon MuiButton-iconSizeMedium"
                      >
                        <span
                          aria-hidden="true"
                          class="material-icons MuiIcon-root"
                        >
                          send
                        </span>
                      </span>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </form>
              </div>
            </div>
            <hr
              class="profile__line"
            />
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-13 profile__exit-button MuiButton-containedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Выйти из аккаунта
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </section>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="app__main main"
    >
      <section
        class="profile container"
      >
        <h2
          class="profile__title visually-hidden"
        >
          Profile page
        </h2>
        <div
          class="profile__inner"
        >
          <div
            class="profile__info info"
          >
            <h4
              class="info__title"
            >
              Информация о пользователе
            </h4>
            <img
              alt="avatar"
              class="info__image"
              height="120"
              src="avatar.png"
              width="120"
            />
            <p
              class="info__name"
            >
              Имя: 
            </p>
            <p
              class="info__age"
            >
              Возраст: 
            </p>
          </div>
          <hr
            class="profile__line"
          />
          <div
            class="profile__settings settings"
          >
            <h4
              class="settings__title"
            >
              Настройки
            </h4>
            <label
              class="MuiFormControlLabel-root settings__online-status"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-3 MuiCheckbox-root Component-root-1"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-6"
                    data-indeterminate="false"
                    name="checkedG"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
              >
                Показывать статус "в сети"?
              </span>
            </label>
            <div
              class="settings__name change-name-form"
            >
              <p
                class="change-name-form__label"
              >
                Изменить имя
              </p>
              <form
                action="#"
                class="change-name-form__input message-form__button"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-8 message-form__input"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-focused Mui-focused Mui-required Mui-required"
                    data-shrink="true"
                    for="outlined-basic"
                    id="outlined-basic-label"
                  >
                    Введите новое имя...
                    <span
                      aria-hidden="true"
                      class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                    >
                       
                      *
                    </span>
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root Mui-focused Mui-focused MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      id="outlined-basic"
                      required=""
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="PrivateNotchedOutline-legendLabelled-11 PrivateNotchedOutline-legendNotched-12"
                      >
                        <span>
                          Введите новое имя...
                           *
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-7 message-form__button MuiButton-containedPrimary"
                  tabindex="0"
                  type="submit"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Отправить
                    <span
                      class="MuiButton-endIcon MuiButton-iconSizeMedium"
                    >
                      <span
                        aria-hidden="true"
                        class="material-icons MuiIcon-root"
                      >
                        send
                      </span>
                    </span>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </form>
            </div>
          </div>
          <hr
            class="profile__line"
          />
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-13 profile__exit-button MuiButton-containedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Выйти из аккаунта
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
